test(storage): add init bucket and filter_reponse

vnvo2409 opened the pull request:
This PR adds:
- Initialization for `Bucket`.
- Some helper functions to support REST `fields` completely. ( Now we can do something like `fields=kind, items(acl(role, entity), name)`)

Related to #4751 

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5166)
<!-- Reviewable:end -->

coryan reviewed (changes requested):
A few requests below.  Let me know what you think.

> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5166#-:-MIKIeUh8VrL9ixxdtUA:b-js1scm)* status: all files reviewed, 2 unresolved discussions (waiting on @vnvo2409)

___
*[google/cloud/storage/emulator/gcs/bucket.py, line 69 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5166#-MIKHqO8EHfd4UkW_jCM:-MIKHqO8EHfd4UkW_jCN:b-ivz9h4) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6464b36fe23dd21fff3b67778a3799f69d978014/google/cloud/storage/emulator/gcs/bucket.py#L69)):*
> ```Python
>             valid &= (
>                 re.match(
>                     "^[0-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}$", bucket_name
> ```

Is this trying to match IP addresses?  Why only IPv4 and not IPv6?  Also, it seems like it would never match because this is only used if `if '.' not in bucket_name:` right?  I would just remove it, we do not need the emulator to be *perfect* we need it to be good enough to help us with testing the libraries.
___
*[google/cloud/storage/emulator/gcs/bucket.py, line 181 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5166#-MIKHFZGCgToKZf_Q-3A:-MIKHFZH4maZO_Mwo6_D:bqch8sl) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6464b36fe23dd21fff3b67778a3799f69d978014/google/cloud/storage/emulator/gcs/bucket.py#L181)):*
> ```Python
>     @classmethod
>     def init_iam_policy(cls, metadata, context):
>         # TODO(vnvo2409): Check for iamConfiguration.uniformBucketLevelAccess.enabled
> ```

Please create a GitHub issue and change this to `TODO(#<issue-number>)`


<!-- Sent from Reviewable.io -->

vnvo2409 commented:
> Is this trying to match IP addresses? Why only IPv4 and not IPv6? Also, it seems like it would never match because this is only used if `if '.' not in bucket_name:` right? I would just remove it, we do not need the emulator to be perfect we need it to be good enough to help us with testing the libraries.

I copied from the old testbench without thiking too much. I removed it in the latest commit.

> Please create a GitHub issue and change this to `TODO(#<issue-number>)`

This case will most likely not happen in the integration test. So I removed it in the lastest commit too.

coryan reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> the build breakage is fixed by #5167. I will wait for that to get fixed before merging these changes.

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5166#-:-MIL98h9Dxg-2s7Z4p4y:b-crww9g)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
